Code/Framework/AzFramework/Platform/Linux/AzFramework/Process/ProcessCommunicator_Linux.cpp

```diff
@@ -38,12 +68,40 @@ namespace AzFramework
     void CommunicatorHandleImpl::Close()
     {
         close(m_handle);
+        if (m_epollHandle != -1)
+        {
+            close(m_epollHandle);
+            m_epollHandle = -1;
+        }
         m_handle = -1;
     }
 
-    void CommunicatorHandleImpl::SetHandle(int handle)
+    bool CommunicatorHandleImpl::SetHandle(int handle, bool createEPollHandle)
     {
         m_handle = handle;
+        // also create an epoll handle
+        if ((handle != -1)&&(createEPollHandle))
```

Comment:
nitpick: whitespace would help with readability here.
```suggestion
        if ((handle != -1) && (createEPollHandle))
```